transcendence-app/src/users/dto/create-user.dto.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  IsEmail,
+  IsInt,
+  IsNotEmpty,
+  IsPositive,
+  IsString,
+  IsUrl,
+} from 'class-validator';
+
+export class CreateUserDto {
```

Comment:
From my point of view, this is a Dto used to create users and make an initial validation, so the name CreateUserDto sounds good to me.

Reply:
Agree but two comments about this. The only place you are using this `userEntity` is, in fact, a param in the main controller methodof `oauth42`, not db related :\

Additionally, a `dto` is itself, as you pointed out, an object which carries data from process to process and just that. Having a `dto` named related to an action lacks of sense because `dto` is only that. If you want to create/update something and pass a dto just create a method for that. In your example, you could do something like:

```typescript
createUserDto(userName: string, password: string) {
    \\ some intermediate logic/validation you may need
    return UserDto(userName, password);
}
```

Reply:
Good point. I've removed the `UserEntity`, and we are using only the `UserDto`. We can add it back later if we need it.